// @streetcred.id/service-clients/dist/custodian/models/mappers.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentEndpoint = {
    serializedName: "AgentEndpoint",
    type: {
        name: "Composite",
        className: "AgentEndpoint",
        modelProperties: {
            did: {
                readOnly: true,
                serializedName: "did",
                type: {
                    name: "String"
                }
            },
            verkey: {
                readOnly: true,
                serializedName: "verkey",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            uri: {
                readOnly: true,
                serializedName: "uri",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.ConnectionContract = {
    serializedName: "ConnectionContract",
    type: {
        name: "Composite",
        className: "ConnectionContract",
        modelProperties: {
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            imageUrl: {
                serializedName: "imageUrl",
                type: {
                    name: "String"
                }
            },
            myDid: {
                serializedName: "myDid",
                type: {
                    name: "String"
                }
            },
            theirDid: {
                serializedName: "theirDid",
                type: {
                    name: "String"
                }
            },
            myKey: {
                serializedName: "myKey",
                type: {
                    name: "String"
                }
            },
            theirKey: {
                serializedName: "theirKey",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            invitation: {
                serializedName: "invitation",
                type: {
                    name: "String"
                }
            },
            invitationUrl: {
                serializedName: "invitationUrl",
                type: {
                    name: "String"
                }
            },
            endpoint: {
                serializedName: "endpoint",
                type: {
                    name: "Composite",
                    className: "AgentEndpoint"
                }
            },
            createdAtUtc: {
                serializedName: "createdAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            multiParty: {
                serializedName: "multiParty",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.CredentialContract = {
    serializedName: "CredentialContract",
    type: {
        name: "Composite",
        className: "CredentialContract",
        modelProperties: {
            credentialId: {
                serializedName: "credentialId",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            definitionId: {
                serializedName: "definitionId",
                type: {
                    name: "String"
                }
            },
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            values: {
                serializedName: "values",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "String"
                        }
                    }
                }
            }
        }
    }
};
exports.ProofAttributeContract = {
    serializedName: "ProofAttributeContract",
    type: {
        name: "Composite",
        className: "ProofAttributeContract",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            value: {
                serializedName: "value",
                type: {
                    name: "String"
                }
            },
            revealed: {
                readOnly: true,
                serializedName: "revealed",
                type: {
                    name: "Boolean"
                }
            },
            selfAttested: {
                serializedName: "selfAttested",
                type: {
                    name: "Boolean"
                }
            },
            conditional: {
                serializedName: "conditional",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.VerificationContract = {
    serializedName: "VerificationContract",
    type: {
        name: "Composite",
        className: "VerificationContract",
        modelProperties: {
            connectionId: {
                serializedName: "connectionId",
                type: {
                    name: "String"
                }
            },
            verificationId: {
                serializedName: "verificationId",
                type: {
                    name: "String"
                }
            },
            state: {
                serializedName: "state",
                type: {
                    name: "String"
                }
            },
            createdAtUtc: {
                serializedName: "createdAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            updatedAtUtc: {
                serializedName: "updatedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            isValid: {
                serializedName: "isValid",
                type: {
                    name: "Boolean"
                }
            },
            verifiedAtUtc: {
                serializedName: "verifiedAtUtc",
                type: {
                    name: "DateTime"
                }
            },
            proof: {
                serializedName: "proof",
                type: {
                    name: "Dictionary",
                    value: {
                        type: {
                            name: "Composite",
                            className: "ProofAttributeContract"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationPolicyParameters = {
    serializedName: "VerificationPolicyParameters",
    type: {
        name: "Composite",
        className: "VerificationPolicyParameters",
        modelProperties: {
            policyName: {
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            credentialId: {
                serializedName: "credentialId",
                type: {
                    name: "String"
                }
            },
            hidden: {
                serializedName: "hidden",
                type: {
                    name: "Boolean"
                }
            }
        }
    }
};
exports.VerificationPolicyRestriction = {
    serializedName: "VerificationPolicyRestriction",
    type: {
        name: "Composite",
        className: "VerificationPolicyRestriction",
        modelProperties: {
            schemaId: {
                serializedName: "schemaId",
                type: {
                    name: "String"
                }
            },
            schemaIssuerDid: {
                serializedName: "schemaIssuerDid",
                type: {
                    name: "String"
                }
            },
            schemaName: {
                serializedName: "schemaName",
                type: {
                    name: "String"
                }
            },
            schemaVersion: {
                serializedName: "schemaVersion",
                type: {
                    name: "String"
                }
            },
            issuerDid: {
                serializedName: "issuerDid",
                type: {
                    name: "String"
                }
            },
            credentialDefinitionId: {
                serializedName: "credentialDefinitionId",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.VerificationPolicyContract = {
    serializedName: "VerificationPolicyContract",
    type: {
        name: "Composite",
        className: "VerificationPolicyContract",
        modelProperties: {
            policyName: {
                serializedName: "policyName",
                type: {
                    name: "String"
                }
            },
            attributeNames: {
                serializedName: "attributeNames",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "String"
                        }
                    }
                }
            },
            availableCredentials: {
                serializedName: "availableCredentials",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "CredentialContract"
                        }
                    }
                }
            },
            restrictions: {
                serializedName: "restrictions",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyRestriction"
                        }
                    }
                }
            }
        }
    }
};
exports.VerificationPoliciesContract = {
    serializedName: "VerificationPoliciesContract",
    type: {
        name: "Composite",
        className: "VerificationPoliciesContract",
        modelProperties: {
            attributes: {
                serializedName: "attributes",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyContract"
                        }
                    }
                }
            },
            predicates: {
                serializedName: "predicates",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "VerificationPolicyContract"
                        }
                    }
                }
            }
        }
    }
};
exports.CustodianWalletContract = {
    serializedName: "CustodianWalletContract",
    type: {
        name: "Composite",
        className: "CustodianWalletContract",
        modelProperties: {
            walletId: {
                serializedName: "walletId",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            }
        }
    }
};
exports.WalletParameters = {
    serializedName: "WalletParameters",
    type: {
        name: "Composite",
        className: "WalletParameters",
        modelProperties: {
            walletId: {
                serializedName: "walletId",
                type: {
                    name: "String"
                }
            },
            ownerName: {
                serializedName: "ownerName",
                type: {
                    name: "String"
                }
            }
        }
    }
};
