// @streetcred.id/service-clients/dist/custodian/models/parameters.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.connectionId = {
    parameterPath: "connectionId",
    mapper: {
        required: true,
        serializedName: "connectionId",
        type: {
            name: "String"
        }
    }
};
exports.credentialId = {
    parameterPath: "credentialId",
    mapper: {
        required: true,
        serializedName: "credentialId",
        type: {
            name: "String"
        }
    }
};
exports.invitation = {
    parameterPath: "invitation",
    mapper: {
        required: true,
        serializedName: "invitation",
        type: {
            name: "String"
        }
    }
};
exports.verificationId = {
    parameterPath: "verificationId",
    mapper: {
        required: true,
        serializedName: "verificationId",
        type: {
            name: "String"
        }
    }
};
exports.walletId0 = {
    parameterPath: "walletId",
    mapper: {
        required: true,
        serializedName: "walletId",
        type: {
            name: "String"
        }
    }
};
exports.walletId1 = {
    parameterPath: [
        "options",
        "walletId"
    ],
    mapper: {
        serializedName: "walletId",
        type: {
            name: "String"
        }
    }
};
