// @streetcred.id/service-clients/dist/agency/models/parameters.js
"use strict";
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.connectionId0 = {
    parameterPath: "connectionId",
    mapper: {
        required: true,
        serializedName: "connectionId",
        type: {
            name: "String"
        }
    }
};
exports.connectionId1 = {
    parameterPath: [
        "options",
        "connectionId"
    ],
    mapper: {
        serializedName: "connectionId",
        type: {
            name: "String"
        }
    }
};
exports.contentType = {
    parameterPath: [
        "options",
        "contentType"
    ],
    mapper: {
        serializedName: "contentType",
        type: {
            name: "String"
        }
    }
};
exports.credentialDefinitionId = {
    parameterPath: "credentialDefinitionId",
    mapper: {
        required: true,
        serializedName: "credentialDefinitionId",
        type: {
            name: "String"
        }
    }
};
exports.credentialId = {
    parameterPath: "credentialId",
    mapper: {
        required: true,
        serializedName: "credentialId",
        type: {
            name: "String"
        }
    }
};
exports.definitionId = {
    parameterPath: "definitionId",
    mapper: {
        required: true,
        serializedName: "definitionId",
        type: {
            name: "String"
        }
    }
};
exports.filename = {
    parameterPath: [
        "options",
        "filename"
    ],
    mapper: {
        serializedName: "filename",
        type: {
            name: "String"
        }
    }
};
exports.messageId = {
    parameterPath: "messageId",
    mapper: {
        required: true,
        serializedName: "messageId",
        type: {
            name: "String"
        }
    }
};
exports.networkId = {
    parameterPath: "networkId",
    mapper: {
        required: true,
        serializedName: "networkId",
        type: {
            name: "String"
        }
    }
};
exports.schemaId = {
    parameterPath: "schemaId",
    mapper: {
        required: true,
        serializedName: "schemaId",
        type: {
            name: "String"
        }
    }
};
exports.state = {
    parameterPath: [
        "options",
        "state"
    ],
    mapper: {
        serializedName: "state",
        type: {
            name: "String"
        }
    }
};
exports.tenantId = {
    parameterPath: "tenantId",
    mapper: {
        required: true,
        serializedName: "tenantId",
        type: {
            name: "String"
        }
    }
};
exports.uploadedFiles = {
    parameterPath: "uploadedFiles",
    mapper: {
        required: true,
        serializedName: "uploadedFiles",
        type: {
            name: "Stream"
        }
    }
};
exports.verificationDefinitionId = {
    parameterPath: "verificationDefinitionId",
    mapper: {
        required: true,
        serializedName: "verificationDefinitionId",
        type: {
            name: "String"
        }
    }
};
exports.verificationId = {
    parameterPath: "verificationId",
    mapper: {
        required: true,
        serializedName: "verificationId",
        type: {
            name: "String"
        }
    }
};
exports.webhookId = {
    parameterPath: "webhookId",
    mapper: {
        required: true,
        serializedName: "webhookId",
        type: {
            name: "String"
        }
    }
};
